src/main/java/fi/aalto/cs/apluscourses/presentation/exercise/SubmissionViewModel.java

```diff
@@ -104,6 +108,25 @@ public String getSubmissionCountText() {
     return submissionCountText.toString();
   }
 
+  /**
+   * Formats a descriptive string for the submission dialog about a submittable file. The string
+   * includes the file name and its last modification date.
+   *
+   * @param file The submittable file in question.
+   */
+  @NotNull
+  public String getFileInformationText(SubmittableFile file) {
```

Comment:
I suggest to consider even more descriptive name for a method, say `getFileNameAndLastModificationTimeText()` or something?

Reply:
I kinda disagree. The contract of this method is to return a human-readable description of the file to be shown in the file list of the dialog. It is this class' choice what exactly is shown there. A subclass could choose to return some other info (as long as it would satisfy the contract).

Reply:
agreed, my bad 🙄